### PR TITLE
Datalake Workload Management

### PR DESCRIPTION
**Pull request by anmolsgandhi**
*Issue #, if available:*

*Description of changes:*

Added Architecture, CFN stacks, python code and shell script for a data lake workload management solution that can be integrated with SDLF stageA and stageB.

Added 2 separate sections; 
1) One for a standalone solution deployment which doesn't require SDLF
2) One for SDLF - WLM integration changes --> Refer to the README.md in sdlf-wlm-integration folder for detailed explanation of changes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+HIGH = os.environ['HIGH']
+LOW = os.environ['LOW']
+ddb = os.environ['ddb'] 
+
+dynamodb_resource = boto3.resource('dynamodb')
+pipeline_table = dynamodb_resource.Table(ddb)
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py:17]: @anmolsgandhi Is there any specific reason as to why the interfaces from the datalake library has not been used?

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+sqs_resource = boto3.resource('sqs')
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py:18]: same as above

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/send-events-sqs/src/lambda_function.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import os
+import urllib.parse
+from botocore.exceptions import ClientError
+import uuid
+import random
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+sns = boto3.client('sns')
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/lambda/send-events-sqs/src/lambda_function.py:12]: Usage of interface from datalake library.

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    -h -- Opens up this help message
+    -p -- AWS profile
+    -r -- AWS Region to deploy to (e.g. eu-west-1)
+    -b -- foundational Bucket name for artifacts
+"; }
+options=':p:r:b:h'
+while getopts $options option
+do
+    case "$option" in
+        p  ) pflag=true; PROFILE=${OPTARG};;
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh:20]: We use -s for the shared cicd profile and -t for the child. Can you please modify this to align with the standard deploy script?

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/cleanup.sh [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+pflag=false
+rflag=false
+bflag=false
+
+DIRNAME=$(pwd)
+
+usage () { echo "
+    -h -- Opens up this help message
+    -p -- AWS profile
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/utils/cleanup.sh:11]: -s for shared cicd account and -t for child account.

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+
+[ -z "$AWS_DEFAULT_REGION" ] && echo "AWS_DEFAULT_REGION is empty" && exit 1
+
+ACCOUNT_NUMBER=`aws sts get-caller-identity --query Account --output text --profile ${PROFILE} --region ${AWS_DEFAULT_REGION}`
+
+if ! $bflag
+then
+    echo "-b not specified, using deafult bucket naming" >&2
+    BUCKET="workload-management-$AWS_DEFAULT_REGION-$ACCOUNT_NUMBER-landing-bucket"
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh:51]: Can you not use the existing artifacts bucket that is a part of the standard deployment.

File: sdlf-utils/pipeline-examples/datalake-workload-management/deploy.sh [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    -h -- Opens up this help message
+    -p -- Name of the AWS profile to use for the Account
+    -r -- AWS Region to deploy to (e.g. eu-west-1)
+    -s -- S3 bucket to store artifact
+"; }
+options=':p:r:s:h'
+while getopts $options option
+do
+    case "$option" in
+        p  ) pflag=true; PROFILE=${OPTARG};;
```
Comment (mousamm) [inline sdlf-utils/pipeline-examples/datalake-workload-management/deploy.sh:18]: -s and -t for profile to align with the existing deploy scripts.

**Review by mousamm — CHANGES_REQUESTED** [review]
Hi @anmolsgandhi ,

Please find my review comments. 

Thanks,
Mousam

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py (hunk 1 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py]: Changed the way repository is structured -- Now contain two sections. sdlf-wlm-integration now uses interfaces and configurations

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py (hunk 2 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/lambda/rerun/src/lambda_function.py]: Changed the way repository is structured -- Now contain two sections. sdlf-wlm-integration now uses interfaces and configurations

File: sdlf-utils/pipeline-examples/datalake-workload-management/lambda/send-events-sqs/src/lambda_function.py (hunk 3 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/lambda/send-events-sqs/src/lambda_function.py]: Changed the way repository is structured -- Now contain two sections. sdlf-wlm-integration now uses interfaces and configurations

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh (hunk 4 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh]: The sdlf-wlm-integration doesnt require any deployment script changes or additions based on new changes. Standalone solution is kept as is

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/cleanup.sh (hunk 5 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/utils/cleanup.sh]: The sdlf-wlm-integration doesnt require any deployment script changes or additions based on new changes. Standalone solution is kept as is

File: sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh (hunk 6 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/utils/bootstrap.sh]: The sdlf-wlm-integration doesnt require any deployment script changes or additions based on new changes. Standalone solution is kept as is

File: sdlf-utils/pipeline-examples/datalake-workload-management/deploy.sh (hunk 7 above)
Comment (anmolsgandhi) [reply sdlf-utils/pipeline-examples/datalake-workload-management/deploy.sh]: The sdlf-wlm-integration doesnt require any deployment script changes or additions based on new changes. Standalone solution is kept as is

**Review by mousamm — CHANGES_REQUESTED** [review]
Can you reuse the json files from the legislators directory for the following files?

sdlf-utils/pipeline-examples/datalake-workload-management/sdlf-wlm-integration/demo/data/memberships.json

sdlf-utils/pipeline-examples/datalake-workload-management/sdlf-wlm-integration/demo/data/organizations.json

sdlf-utils/pipeline-examples/datalake-workload-management/sdlf-wlm-integration/demo/data/persons.json

sdlf-utils/pipeline-examples/datalake-workload-management/sdlf-wlm-integration/demo/data/regions.json

**Pull request merged**